### ISSUE TITLE
`EventLoopError::ExitFailure(code)` and `PumpStatus::Exit(code)` shouldn't use error codes

### ISSUE THREAD

**Issue by daxpedda** [rust-windowing/winit#3059]
After #3056, we should probably not use error codes anymore.

All uses of `EventLoopError::ExitFailure(i32)` could return `EventLoopError::Os(OsError)` instead. `PumpStatus::Exit(i32)` could probably be `PumpStatus::Exit(Result<(), OsError>)` instead, unless we want to add another variant for that.

**Comment by kchibisov** [comment]
Yeah, having exit code is not a good idea, it's better to propagate error and have a dedicated `exit` method, though, some may desire a way to exit event loop with their error somehow, or somehow indicate that loop has terminated.

**Comment by madsmtm** [comment]
But are they exiting the event loop with an error to communicate it to the operating system? Because if so, then they can just do:
```rust
let mut error_exit_code: Option<i32> = None;
event_loop.run(|event, elwt| {
    // On some error event:
    error_exit_code = Some(1);
    elwt.exit()
})?;

if let Some(exit_code) = error_exit_code {
    return exit_code;
} else {
    return 0;
}
```

**Comment by kchibisov** [comment]
Hm, yeah, they can sort of stash like that error, that's true and then just propagate it... Then I guess it's fine...
